bundledata.go

```diff
@@ -168,6 +173,13 @@ func (d *BundleData) UnmarshaledWithServices() bool {
 	return d.unmarshaledWithServices
 }
 
+// SaasSpec represents a single software as a service (SAAS) node.
+// This will be mapped to consuming of offers from a bundle deployment.
+type SaasSpec struct {
+	Source string `bson:",omitempty" json:",omitempty" yaml:",omitempty"`
```

Comment:
What is the purpose of `Source`? I am not sure whether we need to distinguish between local/remote controller offers; when you `juju consume` the client provides a macaroon and the controller details for the one exposing the offer so it shouldn't make any difference (I think).